doc: releases: clean up entry order

danieldegrasse opened the pull request:
Clean up order of entries in the migration guide and release notes to be alphabetical. Combine two duplicate regulator sections, and move driver sections mistakenly placed under networking where they ought to be.

Also, add zephyr-keep-sorted entries to migration guide and release notes to keep alphabetical order enforced.

Migration guide rendered from this PR: https://builds.zephyrproject.io/zephyr/pr/93290/docs/releases/migration-guide-4.2.html

danieldegrasse committed:
doc: releases: migration-guide: clean up entry order

Clean up order of entries in the migration guide to be alphabetical.
Combine two duplicate regulator sections, and move driver sections
mistakenly placed under networking where they ought to be.

Signed-off-by: Daniel DeGrasse <ddegrasse@tenstorrent.com>

pdgendt commented:
This can be forced with

```rst
.. zephyr-keep-sorted-start re(^\w)

Title
=====

.. zephyr-keep-sorted-stop
```

Only lines that start with a word are taken into account.

danieldegrasse committed:
doc: releases: enforce sorting of entries

Enforce sorting of entries, using zephyr-keep-sorted. This will make
sure that entries don't get out of order again within the release notes
or migration guide by throwing a compliance error.

Note that autogenerated areas do not have this enforcement enabled.

Signed-off-by: Daniel DeGrasse <ddegrasse@tenstorrent.com>

danieldegrasse committed:
doc: releases: 4.2: clean up entry order

Clean up order of entries to be alphabetically sorted in the "new apis"
section

Signed-off-by: Daniel DeGrasse <ddegrasse@tenstorrent.com>

danieldegrasse commented:
> Great idea, thank you. Going to update PR with this, so that this gets pulled into future migration guides as well :)

Should be updated now- added the same sorting to release notes, where it caught several things to sort

dleach02 commented on doc/releases/migration-guide-4.2.rst:
```diff
@@ -164,6 +162,24 @@ Display
   the format expected by Zephyr. This change ensures proper behavior of both
   display and video capture samples.
 
+Eeprom
```
This should be all caps since it is an acronym for "electrically erasable programmable read-only memory"
